src/views/ProjectView/ProjectView.tsx

```diff
@@ -32,28 +40,43 @@ const ProjectView = () => {
 
   useEffect(() => {
     if (projectId) {
+      dispatch(setMode('edit'));
       dispatch(setLoading({ text: 'Loading project', id: LOADING_PROJECT }));
       dispatch(getProjectThunk(projectId))
         .then((res) => res.type.includes('rejected') && navigate('/not-found'))
         .catch(Promise.reject)
         .finally(() => dispatch(clearLoading(LOADING_PROJECT)));
-    } else {
+    } else if (projectMode !== 'new') {
       navigate('/planning');
     }
-  }, [projectId]);
+    if (projectMode === 'new') {
```

Comment:
Would this make more sense?

else if(projectMode === 'new')
{
dispatch(resetProject());
}
else
{
 navigate('/planning');
}